daemon/daemon.go

```diff
@@ -542,6 +542,9 @@ func (d *Daemon) init() error {
 	ipv4GW := nodeaddress.GetInternalIPv4()
 	fmt.Fprintf(fw, "#define IPV4_GATEWAY %#x\n", byteorder.HostSliceToNetwork(ipv4GW, reflect.Uint32).(uint32))
 
+	publicIP4 := nodeaddress.GetExternalIPv4()
+	fmt.Fprintf(fw, "#define IPV4_PUBLIC_IP %#x\n", byteorder.HostSliceToNetwork(publicIP4, reflect.Uint32).(uint32))
```

Comment:
Yes, the external IPv4 address is always determined, regardless of whether we allocate IPv4 addresses to endpoints.